pilot/pkg/serviceregistry/kube/controller/ambient/policies.go

```diff
@@ -84,6 +91,46 @@ func PolicyCollections(
 		AuthzDerivedPolicies,
 		PeerAuthDerivedPolicies,
 		DefaultPolicy.AsCollection(),
+		ImplicitWaypointPolicies,
 	}, krt.WithName("Policies"))
 	return AuthzDerivedPolicies, Policies
 }
+
+func implicitWaypointPolicyName(waypoint *Waypoint) string {
+	if !features.DefaultAllowFromWaypoint || waypoint == nil || len(waypoint.ServiceAccounts) == 0 {
+		return ""
+	}
+	// use '_' character since those are illegal in k8s names
+	return "istio_allow_waypoint_" + waypoint.Namespace + "_" + waypoint.Name
+}
+
+func implicitWaypointPolicy(ctx krt.HandlerContext, waypoint Waypoint) *model.WorkloadAuthorization {
```

Comment:
why implicit, we should always allow waypoint to ztunnel

Reply:
That is what the implicit policy does

Reply:
What is implicit ? Maybe it means something different for native English speakers :-)

It seems we explicitly allow some (not clear) workloads to send traffic. How we know they are 'waypoint' or that they are the exact waypoints that we want - not clear, the authorization is based on service accounts, so technically anything that is using some SA will also be able to connect ( implicitly ?).

As this is pretty critical security - would be good to be clear...

Reply:
This was covered in the 'capture' doc fairly thoroughly as part of Status handling.

- A Waypoint SHOULD report its SA on Gateway.Status. This is needed for conformance when we offer alternate waypoint impl support.
- It is recommended that vendors use a waypoint-per-tenant model by default and SHOULD NOT share waypoint SAs with workload SAs. If vendors decide to stray from either of these recommendations they MUST document it and the security implications of doing so.